src/Compilers/CSharp/Portable/Lowering/SpillSequenceSpiller.cs

```diff
@@ -666,6 +667,107 @@ public override BoundNode DefaultVisit(BoundNode node)
 
         #region Expression Visitors
 
+        private new BoundNode VisitLoweredSwitchExpressionArm(BoundLoweredSwitchExpressionArm node)
+        {
+            var statements = VisitList(node.Statements);
+            BoundSpillSequenceBuilder valueBuilder = null;
+            var value = VisitExpression(ref valueBuilder, node.Value);
+            if (valueBuilder is null && statements == node.Statements)
+            {
+                return node.Update(node.Locals, statements, value);
+            }
+
+            var builder = new BoundSpillSequenceBuilder(node.Syntax);
+            builder.AddStatements(statements);
+            builder.Include(valueBuilder);
+            return builder.Update(value);
+        }
+
+        public override BoundNode VisitLoweredSwitchExpression(BoundLoweredSwitchExpression node)
+        {
+            var statements = VisitList(node.Statements);
+            Debug.Assert(statements == node.Statements);
+
+            int firstSpill = -1;
+            BoundSpillSequenceBuilder firstSpillNode = null;
+
+            var switchArms = node.SwitchArms;
+            for (int i = 0; i < switchArms.Length; i++)
```

Comment:
This could be simplified using VisitList if,
- LoweredArm inherits from BoundExpression even though it's a concrete part of the LoweredSwitch.
- Use `VisitList(StaticCast<BoundNode>.From(node.SwitchArms))` which is somehow unconventional.